### PR TITLE
feat: composer exact semver support (without lockfile updates)

### PR DESCRIPTION
**Pull request by kayoub5**
This PR adds the packagist datasource plus basic exact semver Composer support. Composer lockfile updating is not yet supported.

### CODE REVIEW

File: lib/manager/composer/extract.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  for (const depType of depTypes) {
+    if (packageJson[depType]) {
+      try {
+        for (const [depName, version] of Object.entries(packageJson[depType])) {
+          if (depName.includes('/')) {
+            deps.push({
+              depName,
+              depType,
+              currentVersion: version.trim().replace(/^/, ''),
+            });
```
Comment (rarkins) [inline lib/manager/composer/extract.js]: We should also recorded the `lockedVersion` here for each dep too, if lockfile exists

**Comment by evilebottnawi** [comment]
Hm, why lock unsupported? Can i help with this?

**Comment by rarkins** [comment]
@evilebottnawi haven't got to it yet. Also, the way it works is a little different to npm/yarn, so it's not an easy copy/paste. Very happy if you are familiar with Composer and can help!

**Comment by evilebottnawi** [comment]
@rarkins let's do basic support and merge this in master and then i implement support lock/custom registers/etc

**Comment by rarkins** [comment]
In npm/yarn we do this:
- write existing package file and lock file to disk 
- manually patch the package file with new version
- run external command to update lock file

So we don’t actually need to run “npm install foo@x.y.z” to get the job done. 

Another good thing is that when we do this, the existing lock file is “reused” and only the parts that need to be updated are updated. 

For Composer, I’m suspecting we may need to call “composer update” one time let dependency to achieve the same thing.

**Comment by rarkins** [comment]
In its current form, this branch will support extracting and upgrading all exact dependencies. Eg from 1.0.0 to 1.1.0.

We need to port the composer semver logic to JS to be able to support range upgrades too

**Comment by evilebottnawi** [comment]
@rarkins it will be easy when we already have basic support and tests to avoid breaking something

**Comment by rarkins** [comment]
Is it really suitable to merge composer support without lockfile though? If so then I’m happy to merge soon with exact version support only.

**Comment by rarkins** [comment]
@evilebottnawi @kayoub5 let me know if you think this is OK to merge. It works for `composer.json` only and for exact (semver) versions only.

Composer support will remain default disabled, and the docs will include a warning that the solution is incomplete and mostly just for development purposes.

**Pull request merged**